controllers/storagecluster/external_ocs_heartbeat.go

```diff
@@ -0,0 +1,82 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	batchv1 "k8s.io/api/batch/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	externalClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:rbac:groups=batch,resources=cronjobs,verbs=get;list;watch;create;update;patch;delete
+
+func (r *StorageClusterReconciler) enableHeartBeatCronJob(instance *ocsv1.StorageCluster, externalClusterClient *externalClient.OCSProviderClient) error {
+	cronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "heartbeat-to-provider",
+			Namespace: instance.Namespace,
+			OwnerReferences: []metav1.OwnerReference{{
+				UID:        instance.UID,
+				APIVersion: instance.APIVersion,
+				Kind:       instance.Kind,
+				Name:       instance.Name,
+			}},
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule: "* * * * *",
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "heartbeat",
+									Image: os.Getenv(providerAPIServerImage),
+									Command: []string{
+										"/usr/local/bin/consumer-heartbeat",
+										fmt.Sprintf("--storageConsumerUID=%v", instance.Status.ExternalStorage.ConsumerID),
+										fmt.Sprintf("--endpoint=%v", instance.Spec.ExternalStorage.StorageProviderEndpoint),
```

Comment:
Everything here can be read from within the job itself. I think it will be better as in the near future the job will have to collect status and report it back which mean it will need much more info then just connection and identity

Reply:
What does reading from job mean over here?

Reply:
It means that the job code will get the input directly from the storagecluster resources (by reading it) and not via command-line arguments.